include/openthread/tcat.h

```diff
@@ -0,0 +1,276 @@
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief
+ *  This file defines the top-level functions for the OpenThread TCAT.
+ *
+ *  @note
+ *   The functions in this module require the build-time feature `OPENTHREAD_CONFIG_BLE_TCAT_ENABLE=1`.
+ *
+ *  @note
+ *   To enable cipher suite DTLS_PSK_WITH_AES_128_CCM_8, MBEDTLS_KEY_EXCHANGE_PSK_ENABLED
+ *    must be enabled in mbedtls-config.h
+ *   To enable cipher suite DTLS_ECDHE_ECDSA_WITH_AES_128_CCM_8,
+ *    MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED must be enabled in mbedtls-config.h.
+ */
+
+#ifndef OPENTHREAD_TCAT_H_
+#define OPENTHREAD_TCAT_H_
+
+#include <stdint.h>
+#include <openthread/message.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @addtogroup api-ble-secure
+ *
+ * @brief
+ *   This module includes functions that implement TCAT communication.
+ *
+ *   The functions in this module are available when TCAT feature
+ *   (`OPENTHREAD_CONFIG_BLE_TCAT_ENABLE`) is enabled.
+ *
+ * @{
+ *
+ */
+
+#define OT_TCAT_MAX_SERVICE_NAME_LENGTH \
+    15 ///< Maximum string length of a UDP or TCP service name (does not include null char).
+
+/**
+ * Represents TCAT TLV types.
+ *
+ */
+typedef enum otTcatTlvType
+{
+    // Command Class General
+    OT_TCAT_TLV_COMMAND               = 0,  ///< TCAT command TLV   -->  !!!! OBSOLETE!!!!
+    OT_TCAT_TLV_RESPONSE_WITH_STATUS  = 1,  ///< TCAT response with status value TLV
+    OT_TCAT_TLV_RESPONSE_WITH_PAYLOAD = 2,  ///< TCAT response with payload TLV
+    OT_TCAT_TLV_RESPONSE_EVENT        = 3,  ///< TCAT response event TLV (reserved)
+    OT_TCAT_TLV_GET_NETWORK_NAME      = 8,  ///< TCAT network name query TLV
+    OT_TCAT_TLV_DISCONNECT            = 9,  ///< TCAT disconnect request TLV
+    OT_TCAT_TLV_PING                  = 10, ///< TCAT ping request TLV
+    OT_TCAT_TLV_GET_DEVICE_ID         = 11, ///< TCAT device ID query TLV
+    OT_TCAT_TLV_GET_EXTENDED_PAN_ID   = 12, ///< TCAT extended PAN ID query TLV
+    OT_TCAT_TLV_PRESENT_PSKD_HASH     = 30, // 16,   ///< TCAT commissioner rights elevation request TLV using PSKd hash
+    OT_TCAT_TLV_PRESENT_PSKC_HASH     = 17, ///< TCAT commissioner rights elevation request TLV using PSKc hash
+    OT_TCAT_TLV_PRESENT_INSTALL_CODE_HASH =
+        31, // 18,   ///< TCAT commissioner rights elevation request TLV using install code
```

Comment:
18 is actually the correct value after the specifications has renumbered.
I kept 31 temporarily to not break compatibility with the present TCAT client. 
This will be repaired with the next commit.

Reply:
fixed